asQ/allatonce/form.py

```diff
@@ -59,6 +61,18 @@ def __init__(self,
 
         self.form = self._construct_form()
 
+    def time_update(self, t=None):
+
+        if t is not None:
+            self.t0.assign(t)
+
+        else:
+            self.t0.assign(self.t0 + self.dt*self.aaofunc.ntimesteps)
+
+        for i in range(self.nlocal_timesteps):
+            self.time[i].assign(self.time[i] + self.dt*self.aaofunc.ntimesteps)
+        return
```

Comment:
Please can you write a test for this? I don't think it will do the right thing if the `t` argument is given.